bugbug/tools/release_notes.py

```diff
@@ -0,0 +1,216 @@
+
+            current_chunk.append(block)
+            current_token_count += block_token_count
+
+        if current_chunk:
+            chunks.append("\n\n".join(current_chunk))
+
+        return chunks
+
+    def shortlist_with_gpt(self, input_text: str) -> str:
+        return self.summarization_chain.run({"input_text": input_text}).strip()
+
+    def generate_commit_shortlist(self, commit_log_list: list[str]) -> list[str]:
+        commit_log_list_combined = "\n".join(commit_log_list)
+        chunks = self.split_into_chunks(commit_log_list_combined)
+        return [self.shortlist_with_gpt(chunk) for chunk in chunks]
+
+    def clean_commits(
+        self, commit_log_list: list[str], keywords: list[str]
+    ) -> list[str]:
+        cleaned_commits = []
+
+        for block in commit_log_list:
+            if (
+                not any(
+                    re.search(rf"\b{keyword}\b", block, re.IGNORECASE)
+                    for keyword in keywords
+                )
+                and not re.search(
+                    r"release\+treescript@mozilla\.org", block, re.IGNORECASE
+                )
+                and not re.search(r"nightly", block, re.IGNORECASE)
+            ):
+                bug_id_match = re.search(r"Bug (\d+)", block, re.IGNORECASE)
+                if not bug_id_match:
+                    continue
+
+                bug_position = re.search(r"Bug \d+.*", block, re.IGNORECASE)
+                if bug_position:
+                    block = bug_position.group(0)
+
+                commit_summary = block
+                cleaned_commits.append(commit_summary)
+
+        return cleaned_commits
+
+    def remove_unworthy_commits(self, summaries: list[str]) -> str:
+        combined_list = "\n".join(summaries)
+        return self.cleanup_chain.run({"combined_list": combined_list}).strip()
+
+    def select_worthy_commits(self, version: str) -> Optional[str]:
+        self.version2 = version
+        self.version1 = self.get_previous_version(version)
+        self.output_file = f"version_summary_{self.version2}.txt"
+
+        logger.info(f"Generating list of commits for version: {self.version2}")
+        url = f"https://hg.mozilla.org/releases/mozilla-release/json-pushes?fromchange={self.version1}&tochange={self.version2}&full=1"
+        response = requests.get(url)
+        response.raise_for_status()
+
+        data = response.json()
+        commit_log_list = []
+
+        for push_data in data.values():
+            changesets = push_data["changesets"]
+
+            for changeset in changesets:
+                desc = changeset["desc"].strip()
+                if desc:
+                    commit_log_list.append(desc)
+
+        if not commit_log_list:
+            return None
+
+        logger.info("Cleaning commit log...")
+        keywords_to_remove = [
+            "Backed out",
+            "a=testonly",
+            "DONTBUILD",
+            "add tests",
+            "disable test",
+        ]
```

Comment:
Why do we need to pass this list as an argument? Would it be better to keep it constant at the module/file level or include it in the `clean_commits()` function?